geopandas/base.py

```diff
@@ -363,14 +363,98 @@ def count_coordinates(self):
         2    1
         3    5
         4    0
-        dtype: int64
+        dtype: int32
 
         See also
         --------
         GeoSeries.get_coordinates : extract coordinates as a :class:`~pandas.DataFrame`
+        GoSeries.count_geometries : count the number of geometries in a collection
         """
         return Series(self.geometry.values.count_coordinates(), index=self.index)
 
+    def count_geometries(self):
+        """
+        Returns a ``Series`` containing the count of geometries in a collection.
+
+        For single-part geometry objects, this is always 1. For collections, like
+        ``MultiPoint`` or ``MultiLineString``, it is the number of parts in the
+        collection.
```

Comment:
Might be worth mentioning what the result is for a GeometryCollection? (i.e. is it the count of direct parts, or is it recursively?)